gordo_components/client/client.py

```diff
@@ -296,25 +298,16 @@ def predict(
               2nd element is a list of error messages (if any) for running the predictions
         """
         # For every endpoint, start making predictions for the time range
-        jobs = asyncio.gather(
-            *[
-                self._predict(endpoint=endpoint, start=start, end=end)
-                for endpoint in self.endpoints
-            ]
-        )
-
-        # Create new event loop and process getting predictions
-        loop = asyncio.get_event_loop()
-        prediction_results: typing.List[PredictionResult] = loop.run_until_complete(
-            jobs
-        )
-
-        # List of tuples where each represents a single target of name, dataframe of predictions
-        return [
-            (pr.name, pr.predictions, pr.error_messages) for pr in prediction_results
-        ]  # type: ignore
+        with ThreadPoolExecutor(max_workers=self.parallelism) as executor:
+            results = list()
+            jobs = executor.map(
+                lambda ep: self.predict_single_endpoint(ep, start, end), self.endpoints
+            )
+            for result in jobs:
+                results.append((result.name, result.predictions, result.error_messages))
+            return results
```

Comment:
This can be a single list-comprehension, maybe nicer?

Reply:
Also, should we then maybe return an iterator `(b for b in blabla)` vs `[b for b in blabla]`